orthoinference/Jenkinsfile

```diff
@@ -0,0 +1,229 @@
+import groovy.json.JsonSlurper
+
+pipeline{
+	agent any
+	
+	stages{
+		stage('Check if upstream builds succeeded'){
+			steps{
+				script{
+					def orthopairsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/Orthopairs/lastBuild/api/json"
+					def orthopairsStatusJson = new JsonSlurper().parseText(orthopairsStatusUrl.getContent())
+					if(orthopairsStatusJson['result'] != "SUCCESS"){
+						error("Most recent Orthopairs build status: " + orthopairsStatusJson['result'])
+					}
+			    	}
+				script{
+					def updateStIdsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/UpdateStableIdentifiers/lastBuild/api/json"
+					def updateStIdsStatusJson = new JsonSlurper().parseText(updateStIdsStatusUrl.getContent())
+					if(updateStIdsStatusJson['result'] != "SUCCESS"){
+						error("Most recent UpdateStableIdentifiers build status: " + updateStIdsStatusJson['result'])
+					}
+				}	
+		    	}
+	    	}
+		stage('Setup: Backup release_current'){
+			steps{
+				script{
+					dir('orthoinference'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
+							sh "gzip -f ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
+						}
+					}
+				}
+			}
+		}			
+		stage('Setup: Build jar file'){
+			steps{
+				script{
+					dir('orthoinference'){
+						sh "mvn clean compile assembly:single"
+					}
+				}
+			}
+		}
+		stage('Main: Infer mmus'){
+			steps {
+				script{
+					dir('orthoinference'){
+						withCredentials([file(credentialsId: 'Config', variable: 'FILE')]){
+							sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/orthoinference-${env.ORTHOINFERENCE_VERSION}-jar-with-dependencies.jar $FILE mmus"
+						}
+					}
+				}
+			}
```

Comment:
I remember you mentioned that it was necessary to repeat the inference stage for each species, but I was just wondering if using a groovy function is possible in the Jenkins file to consolidate the common parts?

Reply:
Yea I tried to do this a couple of weeks ago. I need to take another stab at it for sure

Reply:
This has been done now, although in a bit of a janky way